resource.go

```diff
@@ -17,6 +17,10 @@ import (
 	"k8s.io/client-go/rest"
 )
 
+const (
+	HeartBeatUpdateThreshold = 30 * time.Minute
```

Comment:
Not deep into this story so I wanted to check this. We're aiming for a heartbeat of 30 mins? The 10 mins mentioned in PR is just for testing.

Also for my interest are we starting with 30 mins and then reducing as we get experience? Or is this the "final" value.

Reply:
That is hard to decide. We get a new event with every change so updating the heartbeat all the time would deadlock itself. So we have to threshold it. That approach feels wrong or incomplete. Maybe the reconciliation has to be improved or maybe the mechanism. Not certain right now. This is how we can make it work now and when there are more strict requirements we have to get together and see what we can do.

Reply:
Ack. Thanks for the explanation. Makes sense to iterate on the threshold as we get more experience with it.